app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt

```diff
@@ -529,6 +534,24 @@ class ChatPresenter(context: Context) :
         computeOtherMessage()
     }
 
+    override fun determinePlanAction(response: SusiResponse?, parseSusiHelper: ParseSusiResponseHelper) {
+        val query = response?.query.toString()
+        identifier = parseSusiHelper.identifier
+        if (query.contains(ALARM, false) || query.contains(PLAN, false)) {
+            // Perform the task thats need to be done after one minute
+            delayIdentifier = identifier
+            planHandler.postDelayed(delayRunnable, 60000) // Time value will change
```

Comment:
Why use handler?

Reply:
I wanted to communicate with the main thread by setting a scheduler. But at the same time, I didn't want the other processes to stop. The user should be able to use the app normally. The scheduler would run in the back ground, and perform the task after the time is over. So I used handler

Reply:
I don't know who has spread this lie that handler tasks tun in background. They don't

They are run on main thread

Reply:
If that is the case, I will surely re factor the code. And write it in background thread seperately.

Reply:
If I am not wrong, handlers provide the way to interact with the main UI thread. Here in this feature our main task is to perform a task, that will interact with ui changes. Like for ```plan to tell a joke in 1 minute``` the scheduler should display a joke in the char after 1 minute. So won't it be good to use handler here?

Reply:
Yes, it will be. But here, you're just statically writing `postDelayed(..., 60000)`

Reply:
The present skills are framed for testing purpose and the scheduler is fixed for 1 minute. No, methods have been implemented in the server to detect the time and send that response. So, I have statically set the delayed to 1 minute. Once, that is implemented in the server, I will change that time accordingly in the client-side.